### PR TITLE
Lowercase the XHR metrics

### PR DESCRIPTION
**Pull request by theninj4**
This PR changes the metrics we emit for XHR requests from this:
```
xhr.timing.POST.www_example_com/foo.UNSENT
```
to this:
```
xhr.timing.post.www_example_com/foo.unsent
```

### CODE REVIEW

File: dist/barometer-test.js [hunk 1]
```diff
@@ -386,7 +386,7 @@ window.XMLHttpRequest = function () {
 
         event(req, 'loadend', function (event) {
           return xhrStats._logMetrics({
-            type: type,
+            type: type.toLowerCase(),
```
Comment (theninj4) [inline dist/barometer-test.js]: This part is the `GET` / `POST` part, which gets passed in by jQuery / superagent / etc when making raw XHR requests.

File: dist/barometer-test.js [hunk 2]
```diff
@@ -329,7 +329,7 @@ var event = require('./event.js')
 var transport = require('./transport.js')
 var urlSanitiser = require('./urlSanitiser.js')
 xhrStats._XMLHttpRequest = window.XMLHttpRequest
-var xhrStates = [ 'UNSENT', 'OPENED', 'HEADERS_RECEIVED', 'LOADING', 'DONE' ]
+var xhrStates = [ 'unsent', 'opened', 'headers_received', 'loading', 'done' ]
```
Comment (theninj4) [inline dist/barometer-test.js]: The states themselves are identified by numeric id, we use this as a mapping to convert a state of `3` to something human readable.

File: dist/barometer-test.js (hunk 1 above)
Comment (pmcnr-hx) [inline dist/barometer-test.js]: Are we sure type will always be a string? No danger of `undefined` or ` null`? Same below.

**Comment by pmcnr-hx** [comment]
Looks good: 👍

File: dist/barometer-test.js (hunk 1 above)
Comment (theninj4) [inline dist/barometer-test.js]: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open
```
method
The HTTP method to use, such as "GET", "POST", "PUT", "DELETE", etc. Ignored for non-HTTP(S) URL
```

Comment (theninj4) [inline dist/barometer-test.js]: I've defaulted this to `unknown` just in case :+1:

**Pull request merged**